File: python/src/sort/quick_sort.py

def quick_sort(in_array):
    if len(in_array) < 2:
        return in_array
    
    pivot=get_pivot(in_array)
    less, greater=[], []

    for i in range(len(in_array)):
        if in_array[i] < pivot:
            less += [in_array[i]]
        else:
            greater += [in_array[i]]
    
    return quick_sort(less) + [pivot] + quick_sort(greater)

def get_pivot(in_array):
    index=len(in_array) >> 1
    return in_array.pop( index )

File: c#/Sort/QuickSort.cs
using System.Collections;
using System.Collections.Generic;

namespace Sort
{
    public class QuickSort : ISort
    {

        private int _pivot = 0;
        public int PivotIndex
        {
            get { return _pivot; }
            set { _pivot = value >> 1; }
        }

        public ArrayList Sort(ArrayList inArray)
        {
            if (inArray.Count < 2)
                return inArray;

            PivotIndex = inArray.Count;
            var pivot = inArray[PivotIndex];
            inArray.RemoveAt(PivotIndex);
            
            ArrayList less = new ArrayList(), greater = new ArrayList();
            for (int i = 0; i < inArray.Count; i++ )
            {
                if ((int)inArray[i] < (int)pivot)
                    less.Add(inArray[i]);
                else
                    greater.Add(inArray[i]);
            }

            ArrayList result = Sort(less);
            result.Add(pivot);
            result.AddRange(Sort(greater));

            return result;
        }
    }
}
